File: library/src/test/java/de/westnordost/osmfeatures/LivePresetDataAccessAdapter.java
package de.westnordost.osmfeatures;

import okio.BufferedSource;
import okio.FileHandle;
import okio.Okio;

import java.io.IOException;
import java.io.InputStream;
import java.net.URL;

import static de.westnordost.osmfeatures.TestUtils.listOf;

public class LivePresetDataAccessAdapter implements FileAccessAdapter
{
    @Override public boolean exists(String name)
    {
        return listOf("presets.json", "de.json", "en.json", "en-GB.json").contains(name);
    }

    @Override public okio.Source open(String name) throws IOException
    {
        URL url;
        if(name.equals("presets.json"))
        {
            url = new URL("https://raw.githubusercontent.com/openstreetmap/id-tagging-schema/main/dist/presets.json");
        } else
            {
            url = new URL("https://raw.githubusercontent.com/openstreetmap/id-tagging-schema/main/dist/translations/"+name);
        }
        return Okio.source(url.openStream());
    }
}

File: library/src/commonTest/kotlin/osmfeatures/LivePresetDataAccessAdapter.kt
package osmfeatures

import java.io.IOException
import java.net.URL
import osmfeatures.TestUtils.listOf
import okio.source

class LivePresetDataAccessAdapter : FileAccessAdapter {
    @Override
    override fun exists(name: String): Boolean {
        return listOf("presets.json", "de.json", "en.json", "en-GB.json").contains(name)
    }

    @Override
    @kotlin.Throws(IOException::class)
    override fun open(name: String): okio.Source {
        val url: URL = if (name == "presets.json") {
            URL("https://raw.githubusercontent.com/openstreetmap/id-tagging-schema/main/dist/presets.json")
        } else {
            URL("https://raw.githubusercontent.com/openstreetmap/id-tagging-schema/main/dist/translations/$name")
        }
        return url.openStream().source()
    }
}
